pandas/core/arrays/base.py

```diff
@@ -1181,10 +1181,6 @@ def convert_values(param):
                     ovalues = [param] * len(self)
                 return ovalues
 
-            if isinstance(other, (ABCSeries, ABCIndexClass)):
-                # rely on pandas to unbox and dispatch to us
-                return NotImplemented
```

Comment:
Personally, I wouldn't use the decorator here. This class is meant to be used by external EA authors (and in that way also serves somewhat as documentation), and it is not used internally (except for tests). 
I find the "easy understandability" in this case more important than the small code duplication (so it is clear to EA authors this method is returning NotImplemented for pandas objects.

And you can add ABCDataFrame to the check to make it correct (and we should have a test to catch this ..)

Reply:
I think its fine to make this change. why have extra-boilerplate code

Reply:
> why have extra-boilerplate code

For the reason I gave in the above comment

Reply:
Can we simply remove it here? (or actually give arguments for why you want to change it)
This is basically a toy/helper implementation that is not used internally. 

(I am completely fine with the rest of the PR)

Reply:
@jbrockmendel I am ok with this here, but I suppose @jorisvandenbossche is stronger here, I guess revert this part.; though this makes things a bit inconsistent across pandas which is odd